version/mkversion/mkversion.go

```diff
@@ -190,7 +190,12 @@ func tailscaleModuleRef(modBs []byte) (string, error) {
 		// Get the last - separated part of req.Mod.Version
 		// (which is the git hash).
 		if i := strings.LastIndexByte(req.Mod.Version, '-'); i != -1 {
-			return req.Mod.Version[i+1:], nil
+			// If the last part is "pre", the version is a pre-release.
```

Comment:
Clarify that this is just a Tailscale convention and not a Go or git thing?

Reply:
Sadly a complete fix needs more.

Specifically this is a go supported thing, go version parsing now supports approximately -[a-zA-Z0-9.] suffixes for prerelease things and considers them less than the version on the left without the suffix.

Unfortunately this collides with our scheme, and with the module scheme.

This should perhaps look for specifically git sha-like values instead, of specific length and hex only.